### PR TITLE
rabbit_core_metrics: Restore compatibility with Erlang R16B03/17.x

### PR DESCRIPTION
**Pull request by dumbbell**
`ets:update_counter/4` was added to Erlang 18.x. Thus we need a wrapper to support Erlang down to R16B03.

This wrapper uses `ets:insert_new/2` first, then `ets:update_counter/3`. It lacks the atomicity of `ets:update_counter/4`, so if the entry was removed in between, the exception is simply ignored.

To avoid any penalty on Erlang 18+, we use `code_version` to load the appropriate version of the wrapper at runtime. Thus we still use `ets:udpate_counter/4` on Erlang 18+.

[#135769137]

### CODE REVIEW

**Comment by dumbbell** [comment]
This change is for `stable` only, it must not be merged to `master`!

File: src/rabbit_core_metrics.erl [hunk 1]
```diff
@@ -130,35 +146,59 @@ channel_stats(reductions, Id, Value) ->
...
+%% ets:update_counter(Tab, Key, Incr, Default) appeared in Erlang 18.x.
+%% We need a wrapper for Erlang R16B03 and Erlang 17.x. This wrapper
+%% first creates the entry if it's missing, then updates it.
+
+ets_update_counter(Tab, Key, Incr, Default) ->
+    code_version:update(?MODULE),
+    ?MODULE:ets_update_counter(Tab, Key, Incr, Default).
+
+ets_update_counter_pre_18(Tab, Key, Incr, Default) ->
+    ets:insert_new(Tab, Default),
```
Comment (dcorbacho) [inline src/rabbit_core_metrics.erl]: I'd insert only if the `update_counter` fails, in the `badarg`, thus avoiding two ETS operations in most cases

Comment (dcorbacho) [inline src/rabbit_core_metrics.erl]: Not strictly.`ets:update_counter/4` always inserts the value if they key is not there, thus `update_counter/4` always succeeds. Now, we are allowing `0`s (let's do nothing) which weren't possible before.

Comment (dumbbell) [inline src/rabbit_core_metrics.erl]: You're right. I changed the patch to use your approach. Thanks!

**Pull request merged**